api/src/main/java/io/opentelemetry/trace/propagation/TracingContextUtils.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.trace.propagation;
+
+import io.grpc.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.ContextUtils;
+import io.opentelemetry.trace.DefaultSpan;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.Tracer;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Util methods/functionality to interact with the {@link io.grpc.Context}.
+ *
+ * <p>Users must interact with the current Context via the public APIs in {@link Tracer} and avoid
+ * accessing this class directly.
+ *
+ * @since 0.1.0
+ */
+@Immutable
+public final class TracingContextUtils {
+  private static final Context.Key<Span> CONTEXT_SPAN_KEY =
+      Context.<Span>key("opentelemetry-trace-span-key");
+  private static final Context.Key<SpanContext> CONTEXT_SPANCONTEXT_KEY =
+      Context.<SpanContext>key("opentelemetry-trace-spancontext-key");
+  private static final Span DEFAULT_SPAN = DefaultSpan.getInvalid();
+
+  /**
+   * Creates a new {@code Context} with the given value set.
+   *
+   * @param span the value to be set.
+   * @return a new context with the given value set.
+   * @since 0.1.0
+   */
+  public static Context withSpan(Span span) {
+    return withSpan(span, Context.current());
+  }
+
+  /**
+   * Creates a new {@code Context} with the given value set.
+   *
+   * @param span the value to be set.
+   * @param context the parent {@code Context}.
+   * @return a new context with the given value set.
+   * @since 0.1.0
+   */
+  public static Context withSpan(Span span, Context context) {
+    return context.withValue(CONTEXT_SPAN_KEY, span);
+  }
+
+  /**
+   * Creates a new {@code Context} with the given {@code SpanContext} set.
+   *
+   * @param spanContext the value to be set.
+   * @return a new context with the given value set.
+   * @since 0.3.0
+   */
+  public static Context withSpanContext(SpanContext spanContext) {
+    return withSpanContext(spanContext, Context.current());
+  }
```

Comment:
Not sure why we need 2 elements in the `Context`

Reply:
Are we suggesting to add always both of them? It is hard to keep them in sync probably.

Reply:
Oh, we are simply putting `spanContext` in `Context.current()`.

Or maybe you are talking about the possibility to have in a given `Context`, at some moment, both `SpanContext` and `Span`? If that's the case, this can happen, but `Span` will be preferred (as mentioned in the Specification).

Reply:
I am not sure that this is the right thing to do, I would stick with only Span in the Context until we prove that it is not possible.

Reply:
Well, at this time we *need* to store `SpanContext` when doing extraction:

```java
// newCtx will contain a SpanContext instance.
Context newCtx = propagator.extract(...);
```

This is also covered in the specification [now](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#determining-the-parent-span-from-a-context) - unless you plan to drop `SpanContext` entirely, and use it always through `DefaultSpan` for this (which can be used a pseudo parent, that is).

Reply:
Let's not add it right now, so that gives me some time to think a bit more. We can for the moment just move what was already available without adding the SpanContext for the moment.